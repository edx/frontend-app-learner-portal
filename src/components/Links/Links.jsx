import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { faFile } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

function Links(props) {
  const { title, links, className } = props;

  const linkItems = links.map(link => (
    <li key={link.href}>
      <FontAwesomeIcon className="mr-2 text-primary" icon={faFile} />
      <a href={link.href} target="_blank" rel="noopener noreferrer">{link.title}</a>
    </li>
  ));

  return (
    <section className={classNames('links', className)}>
      <h3 className="mb-3">{title}</h3>
      <nav>
        <ul className="list-unstyled">
          {linkItems}
        </ul>
      </nav>
    </section>
  );
}

Links.propTypes = {
  title: PropTypes.string.isRequired,
  links: PropTypes.arrayOf(PropTypes.shape({
    title: PropTypes.string.isRequired,
    href: PropTypes.string.isRequired,
  })).isRequired,
  className: PropTypes.string,
};

Links.defaultProps = {
  className: undefined,
};

export default Links;
