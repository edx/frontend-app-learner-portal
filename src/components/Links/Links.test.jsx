import React from 'react';
import { shallow } from 'enzyme';

import Links from './Links';

const linksData = {
  id: 'examples',
  title: 'Examples',
  links: [...new Array(8)].map((_, index) => ({
    title: `Test ${index + 1}`,
    href: `http://example.com/${index + 1}`,
  })),
};

describe('<Links />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow((
      <Links id={linksData.id} title={linksData.title} links={linksData.links} />
    ));
  });

  it('renders at most 5 links by default', () => {
    const listItems = wrapper.find('li');
    expect(listItems).toHaveLength(5);
    expect(wrapper.find('.toggle-show-all-btn').exists()).toBeTruthy();
  });

  it('renders all links when toggled', () => {
    wrapper.find('.toggle-show-all-btn').simulate('click');
    const listItems = wrapper.find('li');
    expect(listItems).toHaveLength(8);
  });
});
