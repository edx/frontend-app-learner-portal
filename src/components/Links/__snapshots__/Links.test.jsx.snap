// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Links /> renders correctly 1`] = `
<section
  className="links"
>
  <h3
    className="mb-3"
  >
    examples
  </h3>
  <nav>
    <ul
      className="list-unstyled"
    >
      <li>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-file fa-w-12 mr-2 text-primary"
          data-icon="file"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 384 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M224 136V0H24C10.7 0 0 10.7 0 24v464c0 13.3 10.7 24 24 24h336c13.3 0 24-10.7 24-24V160H248c-13.2 0-24-10.8-24-24zm160-14.1v6.1H256V0h6.1c6.4 0 12.5 2.5 17 7l97.9 98c4.5 4.5 7 10.6 7 16.9z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        <a
          href="https://example1.com"
          rel="noopener noreferrer"
          target="_blank"
        >
          I am first
        </a>
      </li>
      <li>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-file fa-w-12 mr-2 text-primary"
          data-icon="file"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 384 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M224 136V0H24C10.7 0 0 10.7 0 24v464c0 13.3 10.7 24 24 24h336c13.3 0 24-10.7 24-24V160H248c-13.2 0-24-10.8-24-24zm160-14.1v6.1H256V0h6.1c6.4 0 12.5 2.5 17 7l97.9 98c4.5 4.5 7 10.6 7 16.9z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        <a
          href="https://example2.com"
          rel="noopener noreferrer"
          target="_blank"
        >
          I am second
        </a>
      </li>
      <li>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-file fa-w-12 mr-2 text-primary"
          data-icon="file"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 384 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M224 136V0H24C10.7 0 0 10.7 0 24v464c0 13.3 10.7 24 24 24h336c13.3 0 24-10.7 24-24V160H248c-13.2 0-24-10.8-24-24zm160-14.1v6.1H256V0h6.1c6.4 0 12.5 2.5 17 7l97.9 98c4.5 4.5 7 10.6 7 16.9z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        <a
          href="https://example3.com"
          rel="noopener noreferrer"
          target="_blank"
        >
          I am third
        </a>
      </li>
    </ul>
  </nav>
</section>
`;
